src/components/Calendar.spec.tsx

```diff
@@ -0,0 +1,141 @@
+import Calendar from './Calendar';
+import { mountWithProviders } from '../utils/testing';
+import { ZetkinEvent } from '../types/zetkin';
+
+describe('Calendar', () => {
+    let dummyDate : Date;
+    let dummyEvents: ZetkinEvent[];
+    const dummyStartTime = '2021-05-10T13:37:00+00:00';
+    const dummyEndTime = '2021-05-10T14:37:00+00:00';
+
+    beforeEach(() => {
+        dummyDate = new Date('May 12, 2021');
+        cy.fixture('dummyEvents.json')
+            .then((data : {data: ZetkinEvent[]}) => {
+                dummyEvents = data.data;
+                dummyEvents[0].start_time = dummyStartTime;
+                dummyEvents[0].end_time = dummyEndTime;
+                dummyEvents[1] = {
+                    ...dummyEvents[0],
+                    'end_time': '2021-05-10T17:37:00+00:00',
+                    'id': 25,
+                    'start_time': '2021-05-10T15:37:00+00:00',
+                };
+            });
+    });
+
+    it('shows seven days of the current week starting on Monday', () => {
+        mountWithProviders(
+            <Calendar events={ dummyEvents } focusDate={ new Date(dummyDate) } />,
+        );
+        cy.get('h3').first().contains('misc.calendar.weeks.mon');
+        cy.get('[data-testid="misc.calendar.weeks.mon"]').contains(10);
+        cy.get('[data-testid="misc.calendar.weeks.tue"]').contains(11);
+        cy.get('[data-testid="misc.calendar.weeks.wed"]').contains(12);
+        cy.get('[data-testid="misc.calendar.weeks.thu"]').contains(13);
+        cy.get('[data-testid="misc.calendar.weeks.fri"]').contains(14);
+        cy.get('[data-testid="misc.calendar.weeks.sat"]').contains(15);
+        cy.get('[data-testid="misc.calendar.weeks.sun"]').contains(16);
+    });
+
+    it('shows events that occur on the specified date', () => {
+        mountWithProviders(
+            <Calendar events={ dummyEvents } focusDate={ new Date(dummyDate) } />,
+        );
+        cy.get('[data-testid="misc.calendar.weeks.mon-events"]').children().should('be.visible');
```

Comment:
This feels like a risky test to me, that is slightly too close to the implementation to be ergonomic. If you change the implementation you will need to change the test too, or it might be a false positive – the worst case of test bug because it can potentially obfuscate bugs.

The reason why I'm saying this is that you are just verifying that the children of the element are visible, not what those children actually are. Right now you have this structure

* Day container
  * Events list (`[data-testid="misc.calendar.weeks.mon-events"]`)
    * Event item (these need to be visible for the test to succeed)

But what if you changed to the following structure down the line, for some (easy to imagine) reason?

* Day container
  * Events list (`[data-testid="misc.calendar.weeks.mon-events"]`)
    * Some container
      * Event item

Unless I'm mistaken, the test will now just verify that _Some container_ is visible, which does not necessarily mean that it contains any events or that those are visible. The test will succeed, even if the events are not there.

I think you should instead look for the actual events, and make sure that it's the events you expect them to be, by looking at some of their content, for instance.